miden-lib/build.rs

```diff
@@ -129,6 +129,31 @@ fn compile_executable_modules(source_dir: &Path, target_dir: &Path) -> io::Resul
     Ok(())
 }
 
+// COMPILE KERNELS
+// ================================================================================================
+
+fn compile_kernels(source_dir: &Path, target_dir: &Path) -> io::Result<()> {
+    for masm_file_path in get_masm_files(source_dir)? {
+        // we don't need to compile the api.masm file because it is included in the miden library
+        if "api.masm" == masm_file_path.file_name().unwrap().to_str().unwrap() {
+            continue;
+        }
+        // read the MASM file, parse it, and serialize the parsed AST to bytes
+        let ast = ProgramAst::parse(&fs::read_to_string(masm_file_path.clone())?)?;
+        let bytes = ast.to_bytes(AstSerdeOptions { serialize_imports: true });
+
+        // TODO: get rid of unwraps
+        let masb_file_name = masm_file_path.file_name().unwrap().to_str().unwrap();
+        let mut masb_file_path = target_dir.join(masb_file_name);
+
+        // write the binary MASM to the output dir
+        masb_file_path.set_extension("masb");
+        fs::write(masb_file_path, bytes)?;
+    }
```

Comment:
Rather than doing this dynamically, we can probably just hard-code the file names for transaction kernel main file. That is, that is:

- The source would be `asm/kernels/transaction/main.masm`.
- The target would be something like `assets/kernels/transaction.masb`.

This also means that the location of source files should be under `asm/kernels/transaction` (not just `asm/kernels`, and the output should be in a sub-directory of `assets` (and we probably should do the same for note scripts to avoid collisions).